# For more details

### More information on de Bruijn graphs:

- [How to apply de Bruijn graphs to genome assembly](https://www.nature.com/articles/nbt.2023)
- [Genome Assembly: An Introduction (2016)](https://figshare.com/articles/presentation/Genome_Assembly_An_Introduction_2016_/2972323/1)
- [From multiple kmers to multi-kmer de Bruijn graph](https://homolog.us/blogs/genome/2012/10/10/multi-kmer-de-bruijn-graphs/)

### More information on SPAdes:

- [SPAdes 3.12.0 Manual](http://cab.spbu.ru/files/release3.12.0/manual.html)
- [SPAdes: A New Genome Assembly Algorithm and Its Applications to Single-Cell Sequencing](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3342519/)
- [SPAdes Video Presentation by Sergey Nikolenko](https://www.youtube.com/watch?v=CRJxG29M_Lc&ab_channel=%D0%9B%D0%B5%D0%BA%D1%82%D0%BE%D1%80%D0%B8%D1%83%D0%BC)

### More information on Velvet and VelvetOptimiser:

- [Velvet: de novo assembly using very short reads](ebi.ac.uk/~zerbino/velvet/velvet_poster.pdf)
- [Using the Velvet de novo assembler for short-read sequencing technologies](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2952100/)
- [Velvet Manual - version 1.1 ](https://www.ebi.ac.uk/~zerbino/velvet/Manual.pdf)
- [VelvetOptimiser](https://github.com/tseemann/VelvetOptimiser)

### More information on _de novo_ assembler comparisons:
- [Assessment of de novo assemblers for draft genomes: a case study with fungal genomes](https://bmcgenomics.biomedcentral.com/track/pdf/10.1186/1471-2164-15-S9-S10.pdf)
- [Benchmarking of de novo assembly tools: SPAdes 3.9 vs Velvet 1.2](https://www.eurl-ar.eu/CustomerData/Files/Folders/34-wgs/537_7-1-engage-benchmarking-of-de-novo-assembly-tools-spades-3-9-vs-velvet-1-2.pdf)
- [GAGE-B: an evaluation of genome assemblers for bacterial organisms](https://academic.oup.com/bioinformatics/article/29/14/1718/229886)
